src/main/java/com/metamx/metrics/HttpPostEmitterMonitor.java

```diff
@@ -65,6 +70,15 @@ public boolean doMonitor(ServiceEmitter emitter)
     return true;
   }
 
+  private void emitTimeCounterMetrics(ServiceEmitter emitter, AtomicTimeCounter timeCounter, String metricNameBase)
```

Comment:
Maybe for the sake of (slightly) faster execution, we should just write out the the entire boilerplate code such as:
```
emitter.emit(builder.build("emitter/batchFilling/timeMsSum", AtomicTimeCounter.timeSum(state)));
emitter.emit(builder.build("emitter/batchFilling/count", AtomicTimeCounter.count(state)));
...
emitter.emit(builder.build("emitter/successfulStanding/timeMsSum", AtomicTimeCounter.timeSum(state)));
```
This is because I am not sure if the compiler will optimize the String concatenation in the `emitTimeCounterMetrics`.

Reply:
It doesn't matter much, firstly, those monitors are not called very often (once a minute, at most), secondly, a lot of allocations are done around this logic, much bigger by total size than those strings.